src/Core/Resolvers/DWSqlQueryBuilder.cs

```diff
@@ -82,21 +82,55 @@ private string BuildSqlQuery(SqlQueryStructure structure)
                                     structure.FilterPredicates,
                                     Build(structure.Predicates),
                                     Build(structure.PaginationMetadata.PaginationPredicate));
-            string columns = WrappedColumns(structure);
-            string orderBy = $" ORDER BY {Build(structure.OrderByColumns)}";
 
-            string query = $"SELECT TOP {structure.Limit()} {columns}"
+            string aggregations = string.Empty;
+            if (structure.GroupByMetadata.Aggregations.Count > 0)
+            {
+                if (structure.Columns.Any())
+                {
+                    aggregations = $",{BuildAggregationColumns(structure.GroupByMetadata)}";
+                }
+                else
+                {
+                    aggregations = $"{BuildAggregationColumns(structure.GroupByMetadata)}";
+                }
+            }
+
+            string query = $"SELECT TOP {structure.Limit()} {WrappedColumns(structure)} {aggregations}"
                 + $" FROM {fromSql}"
-                + $" WHERE {predicates}"
-                + orderBy;
+                + $" WHERE {predicates}";
+
+            // Add GROUP BY clause if there are any group by columns
+            if (structure.GroupByMetadata.Fields.Any())
+            {
+                query += $" GROUP BY {string.Join(", ", structure.GroupByMetadata.Fields.Values.Select(c => Build(c)))}";
+            }
+
+            if (structure.GroupByMetadata.Aggregations.Count > 0)
+            {
+                List<Predicate>? havingPredicates = structure.GroupByMetadata.Aggregations
+                      .SelectMany(aggregation => aggregation.HavingPredicates ?? new List<Predicate>())
+                      .ToList();
+
+                if (havingPredicates.Any())
```

Comment:
as per line 111, havingPredicates is nullable, is it ok to deference it here if it turns out to be null?